config/confighttp/confighttp.go

```diff
@@ -419,6 +429,11 @@ func (hss *ServerConfig) ToServer(_ context.Context, host component.Host, settin
 		o.apply(serverOpts)
 	}
 
+	if hss.AllowH2CUpgrade {
+		h2s := &http2.Server{IdleTimeout: 0}
```

Comment:
I feel like this could surprise users out there. HTTP/2 requires TLS, and without an explicit request from admins to disable TLS, we shouldn't be just accepting insecure connections.

Reply:
Ok so, what do you think about setting the option fo `false` by default and adding a warning message here?